fields/types/list/ListType.js

```diff
@@ -0,0 +1,186 @@
+var FieldType = require('../Type');
+var util = require('util');
+var utils = require('keystone-utils');
+
+var isReserved = require('../../../lib/list/isReserved');
+
+/**
+ * List FieldType Constructor
+ * @extends Field
+ * @api public
+ */
+function list (keystoneList, path, options) {
+	this._underscoreMethods = ['format'];
+	list.super_.call(this, keystoneList, path, options);
+}
+list.properName = 'List';
+util.inherits(list, FieldType);
+
+function validateFieldType (field, path, type) {
+	var Field = field.list.keystone.Field;
+	if (!(type.prototype instanceof Field)) {
+		// Convert native field types to their default Keystone counterpart
+		if (type === String) {
+			type = Field.Types.Text;
+		} else if (type === Number) {
+			type = Field.Types.Number;
+		} else if (type === Boolean) {
+			type = Field.Types.Boolean;
+		} else if (type === Date) {
+			type = Field.Types.Datetime;
+		} else {
+			throw new Error(
+				'Unrecognised field constructor for nested schema path `' + path
+				+ '` in `' + field.list.key + '.' + field.path + '`: ' + type
+			);
+		}
+	}
+	return type;
+}
+
+/**
+ * Registers the field on the List's Mongoose Schema.
+ *
+ * @api public
+ */
+list.prototype.addToSchema = function () {
+	var field = this;
+	var mongoose = this.list.keystone.mongoose;
+
+	var fields = this.fields = {};
+	var fieldsArray = this.fieldsArray = [];
+	var fieldsSpec = this.options.fields;
+	var itemSchema = new mongoose.Schema();
+
+	if (typeof fieldsSpec !== 'object' || !Object.keys(fieldsSpec).length) {
+		throw new Error(
+			'List field ' + field.list.key + '.' + field.path
+			+ ' must be configured with `fields`.'
+		);
+	}
+
+	function createField (path, options) {
```

Comment:
Can't this somehow be the same function that instantiates "regular" FieldTypes? I think DRY should apply here, also to make recursive types sane.